librz/util/subprocess.c

```diff
@@ -1096,6 +1141,13 @@ static RzSubprocessWaitReason subprocess_wait(RzSubprocess *proc, ut64 timeout_m
 	bool child_dead = false;
 	bool timedout = true;
 	bool bytes_enabled = n_bytes != 0;
+
+	/* Check if stdout and stderr are connected to a PTY
+	If yes, then set true if we n_bytes == 0,
+	i.e. waiing for subprocess to end */
+	bool stdout_pty = proc->stdout_fd == proc->master_fd && n_bytes == 0;
+	bool stderr_pty = proc->stderr_fd == proc->master_fd && n_bytes == 0;
```

Comment:
1. `proc->master_fd` is set to -1 for when the subprocess is not in a new PTY.
2. Because `n_bytes == 0` only when we are waiting for the process to die (`n_bytes` are 0, when the caller is `rz_subprocess_wait`). I'll add a comment explaining this in the code.

Reply:
This can be considered a sort of a hack-ish way to figure out whether we are waiting for the subprocess to die or not. The significance of `n_bytes` being zero is also stated in the function's (`subprocess_wait`) Doxygen:

```
 * ... If \p n_bytes is
 * not 0, the function terminates as soon as \p n_bytes bytes have been read or
 * the timeout expires.
```

Reply:
Wouldn't this work well also without the `n_bytes == 0`? IN `read_to_strbuf` `read` would return EIO only when EOF.

Reply:
Fair enough.